presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/FileResourceGroupConfigurationManager.java

```diff
@@ -0,0 +1,242 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.execution.resourceGroups;
+
+import com.facebook.presto.SessionRepresentation;
+import com.fasterxml.jackson.annotation.JsonAnySetter;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Throwables;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.airlift.units.DataSize;
+
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import javax.inject.Provider;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Objects.requireNonNull;
+
+public class FileResourceGroupConfigurationManager
+        implements ResourceGroupConfigurationManager, Provider<List<? extends ResourceGroupSelector>>
+{
+    private final List<ResourceGroupSpec> rootGroups;
+    private final List<? extends ResourceGroupSelector> selectors;
+
+    @Inject
+    public FileResourceGroupConfigurationManager(ResourceGroupConfig config, ObjectMapper mapper)
+    {
+        requireNonNull(config, "config is null");
+        requireNonNull(mapper, "mapper is null");
+
+        File file = new File(config.getConfigFile());
+        ManagerSpec managerSpec;
+        try {
+            managerSpec = mapper.readValue(file, ManagerSpec.class);
+        }
+        catch (IOException e) {
+            throw Throwables.propagate(e);
+        }
+        this.rootGroups = managerSpec.getRootGroups();
+
+        ImmutableList.Builder<ResourceGroupSelector> selectors = ImmutableList.builder();
+        for (SelectorSpec spec : managerSpec.getSelectors()) {
+            selectors.add(new StaticSelector(spec.getUserRegex(), spec.getSourceRegex(), spec.getSessionPropertyRegexes(), spec.getGroup()));
+        }
+        this.selectors = selectors.build();
+    }
+
+    @Override
+    public void configure(ConfigurableResourceGroup group, SessionRepresentation session)
```

Comment:
I'll do this in a separate PR, and refactor the `ResourceGroupSelector` interface at the same time